libbeat/processors/dns/config.go

```diff
@@ -84,9 +84,12 @@ type CacheConfig struct {
 // CacheSettings define the caching behavior for an individual cache.
 type CacheSettings struct {
 	// TTL value for items in cache. Not used for success because we use TTL
-	// from the DNS record.
+	// from the DNS record or the minimum configured TTL.
 	TTL time.Duration `config:"ttl"`
 
+	// Minimum TTL value for successful DNS responses.
+	MinTTL time.Duration `config:"ttl.min" validate:"min=1"`
```

Comment:
YAML cannot support having both a `ttl` and a `ttl.min` because `min` becomes a child to `ttl`.
```suggestion
	MinTTL time.Duration `config:"min_ttl" validate:"min=1"`
```